src/streamlink/stream/streamprocess.py

```diff
@@ -62,19 +66,22 @@ def is_usable(cls, session):
         raise NotImplementedError
 
     def open(self):
-        process = self.spawn(self.parameters, self.arguments)
+        if self.is_usable(self.session):
+            process = self.spawn(self.parameters, self.arguments)
 
-        # Wait 0.5 seconds to see if program exited prematurely
-        time.sleep(0.5)
+            # Wait 0.5 seconds to see if program exited prematurely
+            time.sleep(0.5)
 
-        if not process.poll() is None:
-            if hasattr(self.stderr, "name"):
-                raise StreamError(("Error while executing subprocess, "
-                                   "error output logged to: {0}").format(self.stderr.name))
-            else:
-                raise StreamError("Error while executing subprocess")
+            if not process.poll() is None:
+                if hasattr(self.stderr, "name"):
+                    raise StreamError(("Error while executing subprocess, "
+                                       "error output logged to: {0}").format(self.stderr.name))
+                else:
+                    raise StreamError("Error while executing subprocess")
 
-        return StreamProcessIO(self.session, process, process.stdout, timeout=self.timeout)
+            return StreamProcessIO(self.session, process, process.stdout, timeout=self.timeout)
+        else:
+            raise StreamError("{0} is not installed or not supported on your system".format(os.path.basename(self.cmd)))
```

Comment:
`_check_cmd` might need an update or a cleanup,
it seems like it won't be triggered anymore at least with a rtmp example

https://github.com/streamlink/streamlink/blob/178d386f2f35a9c8d934690cc68f813a5cd77d9c/src/streamlink/stream/streamprocess.py#L163-L172

old

```
$ streamlink --rtmp-rtmpdump vlcc rtmp://localhost:33333
[cli][error] Try 1/1: Could not open stream <RTMPStream({'rtmp': 'rtmp://localhost:33333', 'flv': '-'}, redirect=False> (Could not open stream: Unable to find `vlcc' command)
error: Could not open stream <RTMPStream({'rtmp': 'rtmp://localhost:33333', 'flv': '-'}, redirect=False>, tried 1 times, exiting

```

new

```
$ streamlink --rtmp-rtmpdump vlcc rtmp://localhost:33333
[cli][error] Try 1/1: Could not open stream <RTMPStream({'rtmp': 'rtmp://localhost:33333', 'flv': '-'}, redirect=False> (Could not open stream: vlcc is not installed or not supported on your system)
error: Could not open stream <RTMPStream({'rtmp': 'rtmp://localhost:33333', 'flv': '-'}, redirect=False>, tried 1 times, exiting
```